plugins/org.eclipse.elk.alg.packing.rectangles/src/org/eclipse/elk/alg/packing/rectangles/RectPackingLayoutProvider.java

```diff
@@ -60,13 +63,46 @@ public void layout(final ElkNode layoutGraph, final IElkProgressMonitor progress
         // Whether the nodes should be expanded to fit the aspect ratio during node expansion.
         // Only effective if nodes are expanded.
         boolean expandToAspectRatio = layoutGraph.getProperty(RectPackingOptions.EXPAND_TO_ASPECT_RATIO);
+        // Whether interactive layout is activ.
+        boolean interactive = layoutGraph.getProperty(RectPackingOptions.INTERACTIVE);
 
         List<ElkNode> rectangles = layoutGraph.getChildren();
         DrawingUtil.resetCoordinates(rectangles);
         DrawingData drawing;
+        
+        if (interactive) {
+            EList<ElkNode> fixedNodes = new BasicEList<>();
+            for (ElkNode elkNode : rectangles) {
+                if (elkNode.hasProperty(RectPackingOptions.DESIRED_POSITION)) {
+                    fixedNodes.add(elkNode);
+                }
+            }
+            for (ElkNode elkNode : fixedNodes) {
+                rectangles.remove(elkNode);
+            }
+            ECollections.sort(fixedNodes, (a, b) -> {
```

Comment:
Why not simply use `Collections.sort`? Avoid the EMF dependencies where possible within ELK.

Reply:
You are right, I will correct this.